pgmpy/factors/FactorSet.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python3
+from pgmpy.factors import Factor
+from functools import reduce
```

Comment:
@abinashpanda No need to import `reduce` from `functools`. It's builtin https://docs.python.org/2/library/functools.html#functools.reduce